app/models/user.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  has_secure_password
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+  validates :name, presence: true, length: { maximum: Settings.LENGTH_50 }
+  validates :email, presence: true, length: { maximum: Settings.LENGTH_255 }, format: { with: VALID_EMAIL_REGEX }, uniqueness: true
```

Comment:
```suggestion
  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
  
  has_secure_password
  
  validates :name, presence: true, length: { maximum: Settings.LENGTH_50 }
  validates :email, presence: true, length: { maximum: Settings.LENGTH_255 }, format: { with: VALID_EMAIL_REGEX }, uniqueness: true
```
Đặt lại thứ tự để clear code hơn